File: ejercicios-py/ejercicio9.py
import random

"""
Genera una lista de entre 1 y 20 números aleatorios
:param min:
:param max:
"""
def generate_random_numbers(min, max):
    numbers = []
    amount = random.randint(1, 20)
    
    for i in range(amount):
        numbers.append(random.randint(min, max))
    
    return numbers

data = generate_random_numbers(1, 100)

print(f"Se generaron {len(data)} números aleatorios entre 1 y 100:")

print(f"Números generados: {', '.join(map(str, data))}")

print(f"El número menor es: {min(data)}")
print(f"El número mayor es: {max(data)}")

File: ejercicios-js/ejercicio9.js
/**
 * Genera una lista de entre 1 y 20 números aleatorios
 * @param min
 * @param max
 */
function generateRandomNumbers(min, max) {
    const numbers = []
    const amount = Math.floor(Math.random() * 20) + 1;

    for (let i = 0; i < amount; i++) {
        numbers.push(Math.floor(Math.random() * (max - min + 1)) + min);
    }
    return numbers;
}

const data = generateRandomNumbers(1, 100);

console.group(`Se generaron ${data.length} números aleatorios entre 1 y 100:`);

console.log(`Números generados: ${data.join(', ')}`);

console.log(`El número menor es: ${Math.min(...data)}`);
console.log(`El número mayor es: ${Math.max(...data)}`);
console.groupEnd();